relay-server/src/actors/upstream.rs

```diff
@@ -554,6 +554,10 @@ impl UpstreamRelay {
             }
         }
 
+        if let Some(partition_key) = request.request.partition_key() {
+            builder.header("X-Sentry-Relay-Shard", partition_key);
```

Comment:
TODO: Decide on the actual name for the header.

Reply:
Going with `X-Sentry-Relay-Shard`. The proposal mentions `X-Relay-Shard`, but other headers we set also start with `X-Sentry-Relay-`.